### ISSUE TITLE
null attributes are written, example target="null"

### ISSUE THREAD

**Issue by titoBouzout** [ryansolid/mobx-jsx#35]
Just found that `null` attributes are written, for cases like 

```jsx
<a href={this.props.href} target={this.props.blank ? '_blank' : null}>{this.props.children}</a>
```

it generates

```jsx
<a href="#/m/account/login/" target="null">Login</a>
```

**Comment by ryansolid** [comment]
Hmm.. it's due to writing to properties and depending they reflect that way.  I usually assign to empty string instead of null. So far that hasn't been an issue. But I'm interested where that falls apart.

**Comment by titoBouzout** [comment]
Off the top of my head it will break CSS/querySelector based on attributes. One can argue you may use classes for this but that's not the point I think. 

In my opinion sounds like a good idea to express developer intent besides anything else. I remember writing many years ago `if(!value) el.removeAttribute(key)` in many projects. React seems to do something similar, they won't write `undefined` and `null` values, but they will write `empty strings` and I think all of us expect empty strings to be written, but we do not expect `undefined` and `null` to be written, this is subtle but makes a difference. I believe there's a lot of logic written with that assumption which will break if you default to empty.

I cant help but remember on a funny note this bug? https://bugzilla.mozilla.org/show_bug.cgi?id=225554
Basically if you did `<img HEIGHT="5" WIDTH="100%" SRC="">` then the browser was(or is?) requesting the same page twice on a given request because an empty src means self.

**Comment by ryansolid** [comment]
React defaults to attributes which has other tradeoffs (like they are always strings and basically break Web Components) whereas I default to properties. That code doesn't actually set the attribute directly so there is no removeAttribute going on. I generally also use properties because they are much more performant.  Which is awkward because then we'd need to wrap every property write in a check/guard and translate from property name to a guess of the attribute name. Compiling in the `|| ""` is more doable. But that breaks down too I think with boolean attributes which are fine with true/false.

There was a time I wouldn't have given this a second thought. But I think it's probably reasonable time to review how all attributes or properties are handled. I had someone in Solid want me to translate all attributes to properties on Web Components mostly that they preferred writing rich properties that looked like attributes.. SVGs conversely are all attribute writes. I do like the simplicity of always properties.

**Comment by ryansolid** [comment]
I'm working on defaulting to attributes. For consistency I think it's the right move. It removes an unnecessary mapping layer and makes it consistent with handling of static values. I'm just special casing WebComponents to always use properties. 

So when complete these sort of issues will be no more.
